homeassistant/components/workday/config_flow.py

```diff
@@ -179,6 +184,8 @@ async def async_step_user(
         errors: dict[str, str] = {}
 
         if user_input is not None:
+            if user_input.get(CONF_COUNTRY, NONE_SENTINEL) == NONE_SENTINEL:
+                user_input[CONF_COUNTRY] = None
```

Comment:
I think we should store the `NONE_SENTINEL` in `self.data` and change to `None` in next step before writing the config entry.